PWGCF/TableProducer/filterCorrelations.cxx

```diff
@@ -548,9 +548,43 @@ struct MultiplicitySelector {
   PROCESS_SWITCH(MultiplicitySelector, processMCGen, "Select MC particle count as multiplicity", false);
 };
 
+struct MultiplicitySetBuilder {
+  Produces<aod::CFMultiplicitySets> output;
+  std::vector<float> multiplicities{};
+
+  enum MultiplicityEstimators : uint8_t {
+    kCentFT0C = BIT(0),
+    kMultFV0A = BIT(1),
+    kMultNTracksPV = BIT(2),
+    kMultNTracksGlobal = BIT(3),
+  };
+
+  O2_DEFINE_CONFIGURABLE(cfgEstimatorBitMask, uint16_t, 0, "BitMask for multiplicity estimators to be included in the CFMultiplicitySet tables.");
+
+  using CollisionWithCents = soa::Join<aod::Collisions, aod::CentFT0Cs, aod::PVMults, aod::FV0Mults, aod::MultsGlobal>;
+  void processEstimators([[maybe_unused]] CollisionWithCents const& cents, aod::CFCollRefs const& cfcollisions)
+  {
+    for (const auto& cfcollision : cfcollisions) {
+      const auto& collision = cfcollision.collision_as<CollisionWithCents>();
+      multiplicities.clear();
+      if (cfgEstimatorBitMask & kCentFT0C)
+        multiplicities.push_back(collision.centFT0C());
+      if (cfgEstimatorBitMask & kMultFV0A)
+        multiplicities.push_back(collision.multFV0A());
+      if (cfgEstimatorBitMask & kMultNTracksPV)
+        multiplicities.push_back(collision.multNTracksPV());
+      if (cfgEstimatorBitMask & kMultNTracksGlobal)
+        multiplicities.push_back(collision.multNTracksGlobal());
+      output(multiplicities);
+    }
+  }
+  PROCESS_SWITCH(MultiplicitySetBuilder, processEstimators, "Process auxiliary multiplicity/centrality estimators", false);
```

Comment:
The workflow will hang if there are not default `process` function
Is it intended to always run this task?

Reply:
If it hangs, it remains to be solved. Most of the time it won't be needed - only in some specific cases we need the extra estimators (e.g. outliers, systematics). By default it should not run.

Reply:
As the task is incorporated to the workflow and there is not `process` function assigned to it most of the times, the workflow will hang. At least it used to be that way
Have you tested locally?

Reply:
It could not be part of FilterCF simply?